### DISCUSSION TITLE
Is it possible to have separate nvim-tree in each tab with its own root etc?

### DISCUSSION THREAD

**Discussion by carljohnson93** [nvim-tree/nvim-tree.lua#1960] [Q&A] [answered]
I often use neovim as my "window manager". I use tabs as workspaces, and so I often have different projects opened in different tabs. Now, I've been using NvimTree for few weeks, but this workflow only for past few days and I noticed that nvim-tree kinda maintains single "instance" (I guess?), meaning that whenever I change nvim-tree's root in one tab it gets changed in every other tab after I re-open nvim-tree there. Is there some option that I couldn't find? Would this behaviour be very hard to change/add option for?

**Comment by alex-courtis** [comment]
Multi-tab handling isn't great: #457

You could try playing around with: `:h nvim-tree.sync_root_with_cwd` and the various `nvim-tree.update_focused_file` options to see if they are affecting the per-tab trees.

If you can identify some specific cases that are causing these issues, it would be fantastic if you could raise a [bug with a minimal configuration](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml) that demonstrates the issues. We can fix specific issues, as they are bugs, not desirable behaviour.

**Reply by carljohnson93** [reply]
Hm, okay. Yeah I've tried playing around with said options but they are generally of no use for my use-cases. Now about specific issues and reproducible config - I have one but I'm sure it's already known cause it's an obvious one.

steps are: nvim -> `:NvimTreeToggle` -> `:tabnew` -> `:NvimTreeToggle` -> cd into some dir with `Ctrl-]` -> `:tabprev` -> `:NvimTreeToggle` -> `:NvimTreeToggle` again and you'll see that now tree is same as in last tab.

This is known right? I just don't want to add garbage issue

**Reply by alex-courtis** [reply]
Fantastic! Please [raise a bug](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml)

It's great to find these sorts of issues... I'm not a user of tabs and thus don't tend to see them.

**Reply by loyd** [reply]
I also use a project-per-tab approach, and `sync_root_with_cwd` works as expected. Every tab has its own independent nvim-tree instance.

One downside I detected: when switching to another tab, all LSP diagnostics disappear in the tree (but not in the global counter).

**Comment by alex-courtis** [comment, marked as answer]
Tracking at
- #2255
